src/reducers/iprove.js

```diff
@@ -33,20 +45,82 @@ const reducer = (state = initialState, action) => {
   let newState = JSON.parse(JSON.stringify(state))
   if (action.path) {
     const [key, ...path] = action.path
-    let { depth, index } = scan_state(newState, path, key)
+    let {
```

Comment:
style with whitespace

Reply:
Don't think all these changes are necessary for the PR